### PR TITLE
Add GetDoc subcommand to typescript completer

### PR DESCRIPTION
**Pull request by puremourning**
This builds on the Clang completer support for GetDoc subcommand, and just
    returns the docuemntation supplied by tsserver in the quickinfo response.

References: 
* https://github.com/Valloric/YouCompleteMe/issues/1653
* https://github.com/Valloric/YouCompleteMe/pull/1694
* https://github.com/Valloric/ycmd/pull/229

### CODE REVIEW

File: ycmd/completers/typescript/typescript_completer.py [hunk 1]
```diff
@@ -247,6 +250,20 @@ def _GetType( self, request_data ):
...
       return responses.BuildDisplayMessageResponse( info[ 'displayString' ] )
 
+  def _GetDoc( self, request_data ):
+    with self._lock:
+      self._Reload( request_data )
+      seq = self._SendRequest( 'quickinfo', {
+        'file':   request_data[ 'filepath' ],
+        'line':   request_data[ 'line_num' ],
+        'offset': request_data[ 'column_num' ]
+      })
```
Comment (micbou) [inline ycmd/completers/typescript/typescript_completer.py]: Space before `)`.

File: ycmd/completers/typescript/typescript_completer.py [hunk 2]
```diff
@@ -247,6 +250,20 @@ def _GetType( self, request_data ):
...
+
+      info = self._ReadResponse( seq )[ 'body' ]
+      message = '{0}\n\n{1}'.format( info[ 'displayString' ],
+                                     info[ 'documentation' ] )
+      return responses.BuildDetailedInfoResponse( message )
+
```
Comment (micbou) [inline ycmd/completers/typescript/typescript_completer.py]: Missing blank line.

File: ycmd/tests/subcommands_test.py [hunk 3]
```diff
@@ -1338,6 +1338,65 @@ def RunCompleterCommand_GetType_HasNoType_TypescriptCompleter_test():
...
+                               command_arguments = ['GetDoc'],
+                               line_num = 29,
+                               column_num = 9,
+                               contents = contents,
+                               filetype = 'typescript',
+                               filepath = filepath )
+
+  eq_( {
+         'detailed_info': '(method) Bar.testMethod(): void\n\n'
+                        + 'Method documentation',
```
Comment (micbou) [inline ycmd/tests/subcommands_test.py]: `+` not needed.

File: ycmd/tests/subcommands_test.py [hunk 4]
```diff
@@ -1338,6 +1338,65 @@ def RunCompleterCommand_GetType_HasNoType_TypescriptCompleter_test():
...
+                               line_num = 31,
+                               column_num = 2,
+                               contents = contents,
+                               filetype = 'typescript',
+                               filepath = filepath )
+
+  eq_( {
+         'detailed_info': 'class Bar\n\n'
+                        + 'Class documentation\n\n'
+                        + 'Multi-line',
```
Comment (micbou) [inline ycmd/tests/subcommands_test.py]: Same as above.

Comment (puremourning) [inline ycmd/tests/subcommands_test.py]: fixed

File: ycmd/tests/subcommands_test.py (hunk 3 above)
Comment (puremourning) [inline ycmd/tests/subcommands_test.py]: fixed

File: ycmd/completers/typescript/typescript_completer.py (hunk 2 above)
Comment (puremourning) [inline ycmd/completers/typescript/typescript_completer.py]: I'm happy to change this. Though it is worth noting that in this particular file, i'm just copying the existing style, so I have updated the whole file. Arguably this is a distracting unrelated change, so I hope Val doesn't mind :)

**Comment by puremourning** [comment]
Thanks for the review @micbou 

All fixed.

**Comment by micbou** [comment]
LGTM.

**Comment by Valloric** [comment]
Nice, thanks! Lots of PRs this weekend, keep them coming! :D

@homu r+
